### PR TITLE
Error handling for submitting signatures

### LINKED ISSUE #412: support 404/not found page and a 500/server page
Implement 404 and an error page with these graphics:
https://static.moveon.org/giraffe/images/error404.svg
(with the language: "Oops -- looks like what you're looking for isn't here! Email us at help@moveon.org so we can help you find your way.")
and
https://static.moveon.org/giraffe/images/error500.svg
with a message "Uh-oh, something went wrong! Go back and try again please."

The error500 message should display in appropriate moments when a server error occurred (that the user should know about)

Perhaps the language needs to be discussed a little more, and we should be ok with also reporting what exactly failed (it doesn't have to be that exact message for all errors).

### PR DESCRIPTION
**Pull request by jparkrr**
Adds tests and refactors signPetition to dispatch error actions on bad things from SQS. We also add this failure to the reducer so an error screen will be shown in this case.

We handle errors in a similar way to https://github.com/MoveOnOrg/mop-frontend/pull/450

closes #412

### CODE REVIEW

File: test/actions/petitionActions.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  let prevSign
+  let prevWritable
+  before(() => {
+    prevSign = Config.API_SIGN_PETITION
+    prevWritable = Config.API_WRITABLE
+    Config.API_SIGN_PETITION = 'https://petitions.example.com/api-v1-signatures'
+    Config.API_WRITABLE = true
+  })
+
+  it('still dispatches success if endpoint returns 500', done => {
```
[the remark this answers is not in the document]
Comment (schuyler1d) [reply test/actions/petitionActions.js]: No, I don't think it should return success here.  The reducer might need to change a bit so you can either give a contextual 500 error?  should we keep the person on the petition page, and just show an error in-line?  or move to the full 500?

**Comment by schuyler1d** [comment]
Obviously, nothing should (or ever will :-) go wrong, but as an example of what `/api-v1-signatures` might return with an error, here's something that came back before I fully configured it correctly:
```
{"Error":{"Code":"InvalidMessageContents","Message":"Invalid binary character '#x1B' was found in the message body, the set of allowed characters is #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF].","Type":"Sender"},"RequestId":"d3b71e62-6c83-5f8b-b2de-4cba1399ffe3"}
```
This is probably even have a 200 response

File: test/actions/petitionActions.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jparkrr) [reply test/actions/petitionActions.js]: I think for now it's ok to move to the full 500 page. Until we have a good idea of what could go wrong and how to recover from it.

**Comment by jparkrr** [comment]
@schuyler1d Looks like it would actually return 400 for that one https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html#API_SendMessage_Errors

Though I can check for the error key too if you want

**Pull request merged**